CONTRIBUTING.md

```diff
@@ -0,0 +1,276 @@
+   prior scoping ticket.
+
+## Developer Guidelines
+We have several guidelines for contributing code, extending to both core and outside contributors. These rules help us 
+streamline work and provide the best possible collaboration experience, so please follow them!
+
+If you have a new idea or want to request an improvement to the current process, you can file a 
+[developer issue here][developer-request].
+
+### Local Development
+
+#### Setup
+Follow the instructions laid out [here][lookit-setup-instructions].
+
+#### <a name="organization"></a> What's the high-level overview?
+Conceptually speaking, "Lookit-API" is a misnomer. Lookit is *not actually one service*, it is three (3) processes that
+act as (5) conceptually distinct systems/applications:
+1. The Django web app/http API process, providing:
+    - A web UI for researchers to design and manage experiments
+    - A web UI for participants to discover experiments
+    - An HTTP/JSON API consumed by built experiments
+2. An offline Celery process (connected by message queue) that builds and deploys experiments (web archives)
+3. An offline Celery process (connected by message queue) that sends emails and does some rudimentary cleanup
+
+The processes listed above also has three major *in-network* service dependencies:
+1. **Postgres** for relational data storage, used by all processes
+2. **RabbitMQ** as a message queue to communicate between process #1 and the two offline task runner processes
+3. **Docker** to provide a container environment for building experiments (used by process #2)
+
+And as if that wasn't enough, three major *out of network* service dependencies:
+1. **Amazon S3** for storing video
+2. **Google Cloud Storage** for storing experiment web archives and other static files
+3. [**Pipe**][pipe] to relay video streamed from experiments to Amazon S3
+
+For the kicker, if you want to debug the video renaming webhook, you'll need to install and run `ngrok`, and [add
+a new webhook target][pipe-add-webhook] in the Pipe web UI with the freshly generated URL. _Yikes_. Thankfully, you
+won't need to do this that often. Still - depending on how much you want to set up and debug locally, you could 
+potentially be running anywhere from 2-7 processes at once to support Lookit development.
+
+##### <a name="debugging"></a> OK this all sounds great - where do I point my debugger?
+
+If you want to debug one of the processes, you can always use PDB or the new `breakpoint` builtin. Alternatively, if
+you're using an IDE like PyCharm, you can set up handy debug profiles with neat conveniences like isolated environment
+variables and push-button functionality.
+
+If you just want to hack on the API or UIs, you only need to boot up Postgres and the web server (process #1):
+```bash
+brew services start postgresql  # Or `service postgresql start` if you're on linux
+./manage.py runserver           # Alternatively, `python manage.py runserver`
+```
```

Comment:
this is mostly true except that sometimes an action taken in the UI, like changing study status or adding a new researcher to a lab, kicks off a celery process (e.g. sending an email to them) and that will hang unless you either adapt to not do that or you run the celery tasks too. It might be worth explaining this in more detail in the setup docs rather than trying to maintain an explanation here.

Reply:
changed the order and language to better reflect what's needed